internal/serialization/compact_test.go

```diff
@@ -0,0 +1,129 @@
+			tc.f(t)
+		})
+	}
+}
+
+func compactBasicTest(t *testing.T) {
+	// ported from: com.hazelcast.internal.serialization.impl.compact.integration.CompactFormatIntegrationTest#testBasic
+	tcx := it.CompactTestContext{
+		T:           t,
+		Serializers: []serialization.CompactSerializer{EmployeeDTOCompactSerializer{}},
+	}
+	tcx.Tester(func(tcx it.CompactTestContext) {
+		ctx := context.Background()
+		v := EmployeeDTO{age: 30, id: 102310312}
+		map1 := it.MustValue(tcx.Client1.GetMap(ctx, tcx.MapName)).(*hz.Map)
+		it.Must(map1.Set(ctx, 1, v))
+		map2 := it.MustValue(tcx.Client2.GetMap(ctx, tcx.MapName)).(*hz.Map)
+		it.AssertEquals(t, it.MustValue(map2.Get(ctx, 1)), v)
+	})
+}
+
+func basicQueryTest(t *testing.T) {
+	// ported from: com.hazelcast.internal.serialization.impl.compact.integration.CompactFormatIntegrationTest#testBasicQuery
+	tcx := it.CompactTestContext{
+		T:           t,
+		Serializers: []serialization.CompactSerializer{EmployeeDTOCompactSerializer{}},
+	}
+	tcx.Tester(func(tcx it.CompactTestContext) {
+		ctx := context.Background()
+		map1 := it.MustValue(tcx.Client1.GetMap(ctx, tcx.MapName)).(*hz.Map)
+		for i := int32(0); i < 100; i++ {
+			it.MustValue(map1.Put(ctx, i, EmployeeDTO{age: i, id: 102310312}))
+		}
+		map2 := it.MustValue(tcx.Client2.GetMap(ctx, tcx.MapName)).(*hz.Map)
+		ks := it.MustValue(map2.GetKeySetWithPredicate(ctx, predicate.SQL("age > 19"))).([]interface{})
+		assert.Equal(t, 80, len(ks))
+	})
+}
+
+func joinedMemberQueryTest(t *testing.T) {
+	// ported from: com.hazelcast.internal.serialization.impl.compact.integration.CompactFormatIntegrationTest#testJoinedMemberQuery
+	tcx := it.CompactTestContext{
+		T:           t,
+		Serializers: []serialization.CompactSerializer{EmployeeDTOCompactSerializer{}},
+	}
+	tcx.Tester(func(tcx it.CompactTestContext) {
+		ctx := context.Background()
+		for i := int32(0); i < 100; i++ {
+			it.MustValue(tcx.M.Put(ctx, i, EmployeeDTO{age: i, id: 102310312}))
+		}
+		it.MustValue(tcx.Cluster.StartMember(ctx))
+		err := tcx.ExecuteScript(ctx, fmt.Sprintf(`
+			var m = instance_2.getMap("%s");
+			var size = m.keySet(com.hazelcast.query.Predicates.sql("age > 19")).size();
+			if (size != 80) {
+				throw new Error("size is " + size);
+			}			
+		`, tcx.MapName))
+		assert.NoError(t, err)
+	})
+}
+
+func clusterRestartTest(t *testing.T) {
+	// ported from: com.hazelcast.internal.serialization.impl.compact.integration.CompactFormatIntegrationTest#testJoinedMemberQuery
+	tcx := it.CompactTestContext{
+		T:           t,
+		Serializers: []serialization.CompactSerializer{EmployeeDTOCompactSerializer{}},
+	}
+	tcx.Tester(func(tcx it.CompactTestContext) {
+		ctx := context.Background()
+		v := EmployeeDTO{age: 30, id: 102310312}
+		k := int64(1)
+		it.MustValue(tcx.M.Put(ctx, k, v))
+		// changeCluster
+		tcx.Cluster.Shutdown()
+		it.MustValue(tcx.Cluster.StartMember(ctx))
+		it.MustValue(tcx.M.Put(ctx, k, v))
+		v2 := it.MustValue(tcx.M.Get(ctx, k))
+		assert.Equal(t, v, v2)
+	})
```

Comment:
Since you cache the schemas generated from the serializers, I believe we won't fetch the schemas from the cluster in any of these tests. Can you confirm this? If this is the case, you might add a test where there are two versions of the schema for the same class and make sure that you can read data written by the other client

Reply:
It seems I've missed something while refactoring the test helper. Updated at: https://github.com/hazelcast/hazelcast-go-client/pull/910/commits/a3d1fad7e21a66cc158a7bc9aba4b6257dcfc9af